pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/tests/Test_TNCuda_Expval.cpp

```diff
@@ -195,8 +210,13 @@ TEMPLATE_TEST_CASE("[PauliZ]", "[MPSTNCuda_Expval]", float, double) {
                 {{0}, {1}, {2}, {0, 1}, {1, 2}, {0, 2}},
                 {{false}, {false}, {false}, {false}, {false}, {false}},
                 {{}, {}, {}, {0.5}, {0.6}, {0.7}});
-            mps_state.append_mps_final_state(cutoff, cutoff_mode);
-            auto m = MeasurementsTNCuda<TensorNetT>(mps_state);
+
+            if constexpr (std::is_same_v<TNDeviceT, MPSTNCuda<double>> ||
+                          std::is_same_v<TNDeviceT, MPSTNCuda<float>>) {
+                tn_state->append_mps_final_state(cutoff, cutoff_mode);
+            }
+
+            auto m = MeasurementsTNCuda<TNDeviceT>(*tn_state);
```

Comment:
in L225 can we use Approx?

Reply:
`Matchers` is more approperiate for inequality check